### ISSUE TITLE
Cross-platform Interwindow Drag and Drop

### ISSUE THREAD

**Issue by louy2** [PistonDevelopers/conrod#726]
For a start:

- [ ] X server [XDnD](https://en.wikipedia.org/wiki/X_Window_selection#XDND)
- [ ] Windows [OLE Drag and Drop](http://www.catch22.net/tuts/drag-and-drop-introduction)
- [ ] OS X [Cocoa Drag and Drop](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/DragandDrop/DragandDrop.html)

**Comment by mitchmindtree** [comment]
We should be able to provide this as something similar to glutin's [`DroppedFile`](https://github.com/PistonDevelopers/conrod/pull/703) or SDL's [`SDL_DropEvent`](https://wiki.libsdl.org/SDL_DropEvent). Perhaps working with files (providing a `PathBuf` as in both those events) is the most generic way of supporting drag and drop cross platform?

Once I land #703, this could be added as a variant to the `backend::RawEvent` (and in turn the `event::Ui` and `event::Widget` enums) so that it's up to the backend to provide the event as is the case for all other events.

**Comment by Boscop** [comment]
Yes, a DroppedFile event would be very useful! Even if it's only supported with the winit backend at first.
